core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/SimplifyCountOverConstant.java

```diff
@@ -112,6 +112,10 @@ private boolean isCountOverConstant(Session session, AggregationNode.Aggregation
             return false;
         }
 
+        if (aggregation.isDistinct()) {
```

Comment:
Is it possible to have another rule which could simplify count distinct over aggregation to be simplified as `BIGINT 1`

Reply:
That makes sense. So, `count(distinct 'abc')` will be replaced with `BIGINT 1` - right?